src/components/homepage/Homepage.jsx

```diff
@@ -16,11 +25,24 @@ const Homepage = () => {
     FetchData();
   }, []);
 
+  function fetchDataBasedOnId(id){
+    console.log(id);
+    async function FetchData() {
+      const movie = await GetMoviesBasedOnId(id).then((resp) => {
+        return resp;
+      });
+      setSelectedMovie(movie);
+    }
+    FetchData();
+  }
+
   return (
-    <MovieContainer>
-      {!movies && <div>Loading...</div>}
-      {movies && <Movie movies={movies} />}
-    </MovieContainer>
+    <MovieContext.Provider value={movieContextValue}>
+      <MovieContainer>
```

Comment:
this should be passed as children to  MovieContext.Provider when we move it to a separate file

Reply:
Done